deploy/addons/ingress/ingress-deploy.yaml.tmpl

```diff
@@ -299,7 +299,7 @@ metadata:
   namespace: ingress-nginx
 spec:
   type: NodePort
-  {{- if eq .IngressAPIVersion "v1"}}
+  {{- if and (eq .IngressAPIVersion "v1") (not .LegacyKubernetesVersion)}}
```

Comment:
I think we could choose a better future-proof , since this field is used by all other addons,

Reply:
I would prefer any other name than legacyKubernetesVersion.
even something as preOneTwentyKuberentes

Reply:
it makes sense (and i like it!) - renamed accordingly